### ISSUE TITLE
Update navigation on www and Search to match learn

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#17307]
![settings___freecodecamp_](https://user-images.githubusercontent.com/985197/40752060-7eb67ee8-6433-11e8-8057-faf6ea239cfd.png)

**Comment by johnkennedy9147** [comment]
I think 'settings ' is better UI, clearer and more obvious for users.
Also think 'home' would be better than clicking freeCodeCamp word and icon which doesn't actually look like a button.

**Comment by QuincyLarson** [comment]
@johnkennedy9147 I agree. We should have both: a user photo and settings link - both of which will point to the settings. Is this something you'd be interested in helping us implement?

**Comment by johnkennedy9147** [comment]
@QuincyLarson I'd be happy to but I've got a bunch on at the moment so might be a few weeks before I get a chance to look at it. I've added it to my to-do list but in the meantime anyone else can pick it up.  I'll add a comment if I start working on it.

**Comment by QuincyLarson** [comment]
@johnkennedy9147 Are you still interested in helping tackle this issue? I think we could add both a "home" button (which would take them to the "welcome page" to the navigation, and a "settings" link to the left of their photo (both the photo and the "settings" link would redirect them to /settings).

**Comment by johnkennedy9147** [comment]
@QuincyLarson see my email - however for this issue I think I saw something being done in another PR which has changed the navs - www, learn and settings are all now different

**Comment by QuincyLarson** [comment]
@johnkennedy9147 Yes - we're trying to make the navigation universal across different services. We will add the settings button to the nav when we do so.

**Comment by schalkneethling** [comment]
@QuincyLarson Hey there, is this issue still up for grabs? Also, are the requirements still as defined in https://github.com/freeCodeCamp/freeCodeCamp/issues/17307#issue-327929096

It seems like it might have changed a bit from reading the conversation above. Thanks!

**Comment by SemperDisco** [comment]
What is the current of this issue? Can I be of any help?

**Comment by QuincyLarson** [comment]
@ahmadabdolsaheb Thanks for your patience. 

We want to build a universal navigation component that exists on every page - including the Guide and Learn.

Rather than add a ton of buttons up top, we want to add a footer that has links to all the various pages currently linked from about.freecodecamp.org. The main problem with having a footer is that the curriculum view has 3 independently scrolling elements.

Would you be interested in taking a crack at designing this? And @SemperDisco We would welcome your help as well if you're interested in collaborating on this.

**Comment by moT01** [comment]
Here's a list of what we might want in a footer.
[about](https://about.freecodecamp.org/)
[support](https://support.freecodecamp.org/)
[donate](https://donate.freecodecamp.org/)
[YouTube](https://www.youtube.com/freecodecamp)
[podcast](https://podcast.freecodecamp.org/)
[medium](https://medium.freecodecamp.org/)
[linkedIn](https://www.linkedin.com/uas/login?session_redirect=https%3A%2F%2Fwww.linkedin.com%2Fschool%2F4831032%2Falumni%2F)
[github](https://github.com/freeCodeCamp/)
[gitter](https://gitter.im/FreeCodeCamp/home)

**Comment by ahmadabdolsaheb** [comment]
here is a link to bootstrap official sticky footer sample.
https://getbootstrap.com/docs/4.1/examples/sticky-footer-navbar/

It is the industry standard to group the links for the footer. so we probably need to think of that.
here is a sample from slack
<img width="1304" alt="screen shot 2018-12-26 at 6 59 48 pm" src="https://user-images.githubusercontent.com/4591597/50450694-90003380-0940-11e9-9368-e61fdfa10757.png">

I suggest finalizing the links that we need to include in the footer and then start working on the footer component.

**Comment by ahmadabdolsaheb** [comment]
Any ideas for resolving the 3 scroller issue  @moT01 @SemperDisco ?
I think if we had only one scroller, the user had to scroll to the bottom of the page anyway to get the footer. Right now, i get a pretty similar behavior. when I have my cursor on a scrolling component and scroll to the end I am able to go where the footer component would be. So I don't think it is a major issue because the behavior is similar.

**Comment by QuincyLarson** [comment]
@ahmadabdolsaheb I like the simplicity of Khan Academy's footer, with it's mission statement and call to action on the side:

freeCodeCamp is a donor-supported tax-exempt 501(c)(3) nonprofit organization (United States Federal Tax Identification Number: 82-0779546)

Our mission: to help people learn to code for free. We accomplish this by creating thousands of videos, articles, and interactive coding lessons - all freely available to the public. We also have thousands of freeCodeCamp study groups around the world.

Donations to freeCodeCamp go toward our education initiatives, and help pay for servers, services, and staff. You can [make a tax-deductible donation here](https://donate.freecodecamp.org).

Columns:

Our Nonprofit
---
[About](https://about.freecodecamp.org)
[Donate](https://donate.freecodecamp.org)
[Shop](https://shop.freecodecamp.org)
[Sponsors](https://sponsors.freecodecamp.org)
[Contact email](mailto:team@freecodecamp.org)

Our Community
---
[Alumni Network](https://www.linkedin.com/school/free-code-camp/people/)
[Forum](https://www.freecodecamp.org/forum)
[Gitter](https://gitter.im/FreeCodeCamp/home)
[GitHub](https://github.com/freeCodeCamp/)
[Support](https://support.freecodecamp.org)
[Code of Conduct](https://code-of-conduct.freecodecamp.org)
[Privacy Policy](https://privacy.freecodecamp.org)
[Terms of Service](https://terms-of-service.freecodecamp.org)

Our Learning Resources
---
[Curriculum](https://learn.freecodecamp.org)
[Guide](https://guide.freecodecamp.org)
[YouTube](https://www.youtube.com/freecodecamp)
[Podcast](https://podcast.freecodecamp.org)
[Twitter](https://twitter.com/freecodecamp)
[Medium](https://medium.freecodecamp.org)
[Instagram](https://instagram.com/freecodecamp)

**Comment by ahmadabdolsaheb** [comment]
Here is a quick pen. 
https://codepen.io/abdolsa/pen/rozGJj?editors=1100
any thoughts? @QuincyLarson @moT01 @SemperDisco

**Comment by QuincyLarson** [comment]
@ahmadabdolsaheb This looks great! Very nice clean design, good use of whitespace, and good responsivity.

**Comment by ahmadabdolsaheb** [comment]
@QuincyLarson @moT01 @Nirajn2311 I have a working footer, and as you mentioned it does affect the user experience when placed under three scrolling divs. here are some solutions. 

1. widening the separating the columns separating the three panels so campers have something to grab onto to scroll the whole page rather than individual panels; 

2. have a an expandable fixed bar at the bottom.

3. have a sticky expandable small footer at the bottom.

4. completely removing the footer on this page.

Let me know about your feedback. thanks

**Comment by moT01** [comment]
There's already a lot going on in this area. More scroll bars just seems like clutter and possible confusion...

I like the expandable option(s) or...

5. embed the footer within the lesson panel?

**Comment by moT01** [comment]
Looks good, I was thinking more of a green background to match the nav, but the gray and green buttons look good... might want to get Quincy's opinion there. And actually, that looks pretty good just putting it at the bottom of the other panels... the scrollbar is hidden until you scroll down - I suppose it might be different on other browsers. It could maybe use some white space or something to separate it from the lesson panels a little. But overall, it looks real good - nice job!

**Comment by vkWeb** [comment]
@QuincyLarson I want to be an active participant in designing the footer but as you know I am presently inactive in the community. Fortunately, I landed here while I was looking at the issues. I want to share some ideas on this. 

@ahmadabdolsaheb Amazing work there 👍. I am a big fan of GitHub's footer! We should do something similar. We should put the full footer only on the main homepage. Few primary links may be put on the authorized user's welcome page. Some screenshots will clear the design I have in mind. 

### Full footer on homepage:
![full footer edited](https://user-images.githubusercontent.com/26724128/50784195-00a93780-12d3-11e9-9689-b8b2ee33efa6.png)
#### We can implement something similar to: 
![github full footer](https://user-images.githubusercontent.com/26724128/50784349-6b5a7300-12d3-11e9-99d4-5f64c243475b.png)
### Footer with primary links on authorized user welcome page:
![footer with primary links](https://user-images.githubusercontent.com/26724128/50784374-7a412580-12d3-11e9-9f55-c3c4328c976e.png)
#### We can implement something similar to:
![github footer with primary links](https://user-images.githubusercontent.com/26724128/50784426-9775f400-12d3-11e9-921e-67705e36ac93.png)

And I am totally against putting footer on the `learn` platform. Before making any UI change we should ask ourselves "As a camper will it improve my learning experience?". Putting footer on `learn` platform will not. Presently, we have a lot of scrollbars and buttons on learn platform and putting footer there will conjust things up more. So, no footer on `learn` is the way we should be going :)

**Comment by ahmadabdolsaheb** [comment]
thanks @vkWeb for the valuable input. That is the industry standard to implement footers. So ideally the footer component will receive some props depending on the page that the user is visiting and output a custom footer. Let's wait for @QuincyLarson's output to see which direction he would like us to pursue.

**Comment by Nirajn2311** [comment]
@ahmadabdolsaheb Nice design of the footer. I was wondering if for the social media part(FB, twitter, medium, Youtube etc.) we could change it to logos.

**Comment by QuincyLarson** [comment]
@ahmadabdolsaheb That looks solid! 

@vkWeb Rather than simplify the footer as GitHub has done, I propose we keep it the same everywhere so people don't have to adapt to the fact that the footer changes from page to page. We want to keep our navigation up top the same on every page, too.

Regarding having the footer on Learn - we do eventually want to do this, but we need to figure out how. As @vkWeb pointed out, with the current multi-pain scrolling, having a footer would be a pain. Let's worry about that later, and focus instead on creating a common component we can use on all the non-learn pages.

@ahmadabdolsaheb Since you've already got a working design of the footer, are you interested in tackling this?
